src/yb/yql/pgwrapper/pg_wrapper.cc

```diff
@@ -349,14 +376,33 @@ Status PgWrapper::Start() {
   }
 
   argv.push_back("-c");
-  // TODO: we should probably load the metrics library in a different way once we let
-  // users change the shared_preload_libraries conf parameter.
+
+  // Gather the default extensions:
+  vector<string> metricsLibs;
+  metricsLibs.push_back("pg_stat_statements");
   if (FLAGS_pg_stat_statements_enabled) {
-    argv.push_back("shared_preload_libraries=pg_stat_statements,yb_pg_metrics,pgaudit,"
-      "pg_hint_plan");
-  } else {
-    argv.push_back("shared_preload_libraries=yb_pg_metrics,pgaudit,pg_hint_plan");
+    metricsLibs.push_back("yb_pg_metrics");
   }
+  metricsLibs.push_back("pgaudit");
+  metricsLibs.push_back("pg_hint_plan");
+
+  // Once again parse the pg conf:
+  vector<string> pgConfLines;
+  if (!FLAGS_ysql_pg_conf_csv.empty()) {
+    RETURN_NOT_OK(ReadCSVValues(FLAGS_ysql_pg_conf_csv, &pgConfLines));
```

Comment:
We already process `FLAGS_ysql_pg_conf_csv`  in `WritePostgresConfig` so this now gets processed twice.
I guess the code works because of the override rules between the two, but I think it might be cleaner to do it just once.

Basically, we now have two ways to set config options:
1. Add them to the auto-generated `ysql_pg.conf` configuration file.
2. Setting them in the `postgres` process shell arguments using the `-c` option.

I think it might not be too complicated to simplify that by just using the conf file for the shared preloaded extensions part (we could do it for the other `-c` options also as a follow-up, but not needed in this diff).
So this is a bit of a less trivial change but we could:
1. Move the logic starting around line `380` (from `// Gather the default extensions:`) to the `WritePostgresConfig(` function.
2. In that function, we can reuse the existing `ReadCSVValues` call instead of re-parsing the pg_config flags.
3. Instead of setting the `-c` options, we just iterate over the `lines` from the config, and, similarly to current code, if they match the `shared_preload_libraries` prefix we merge the pre-computed metricsLibs list before proceeding to write them to the final config.

Note: I might have missed some details/complications here so let me know if this change seems a bit involved. If so I can double-check and provide more details or maybe a better suggestion.

Reply:
As in, completely skip the `shared_preload_libraries` from the `-c` switches and move it to the config file? That would work. If you confirm this is the way to go, I can do that.

Reply:
@radekg Correct -- I think moving `shared_preload_libraries` to just the config file will simplify and consolidate our logic/code there.

Reply:
Hey @m-iancu, implemented as requested: https://github.com/yugabyte/yugabyte-db/pull/9576/commits/34c9330fbb0667d7193051596816e0d219486bfc.

Reply:
Thanks @radekg, looks great.  Another small follow-up to this. The user config lines will be few, but there are a lot of lines in the config sample file (all commented out) so it'd be better if we skip iterating over those during processing (to search-for any `shared_preload_libraries` entries and later to remove any duplicates).

I think this patch might work for that but I did not test thoroughly.
```
+  vector<string> user_configs;
   if (!FLAGS_ysql_pg_conf_csv.empty()) {
-    RETURN_NOT_OK(ReadCSVValues(FLAGS_ysql_pg_conf_csv, &lines));
+    RETURN_NOT_OK(ReadCSVValues(FLAGS_ysql_pg_conf_csv, &user_configs));
   } else if (!FLAGS_ysql_pg_conf.empty()) {
-    ReadCommaSeparatedValues(FLAGS_ysql_pg_conf, &lines);
+    ReadCommaSeparatedValues(FLAGS_ysql_pg_conf, &user_configs);
   }

   // If the user has given any shared_preload_libraries, merge them in.
-  for (string &value : lines) {
+  for (string &value : user_configs) {
     if (boost::starts_with(value, "shared_preload_libraries")) {
       MergeSharedPreloadLibraries(value, &metricsLibs);
+    } else {
+      lines.push_back(value);
     }
   }

-  // Ensure we do not have the line duplicated in the config.
-  lines.erase(std::remove_if(lines.begin(), lines.end(), [](const std::string& s) {
-    return boost::starts_with(s, "shared_preload_libraries");
-  }));
-
```
Note:  I (re)started a full run of our tests based on your current patch plus this change, so testing should be done by tomorrow (hopefully everything is good there).

Reply:
Added your diff: https://github.com/yugabyte/yugabyte-db/pull/9576/commits/6ba7647eec280ecaa35871ba707e205822347dc0.